src/Log/Migrations/CreateNewLogTable.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Give\Log\Migrations;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+
+
+/**
+ * Class CreateNewLogTables
+ * @package Give\Log\Migrations
+ *
+ * @since 2.9.7
+ */
+class CreateNewLogTable extends Migration {
```

Comment:
Please use the new `check()` method on these migrations.

Reply:
You mean I should check if the previous migration was successful so that I can run the current one?

Maybe it will be a good idea to have an array of migration dependencies?
For example, if I want to run a migration that depends on other migrations, it will be super useful and easy to just define them as dependencies ( array of migration IDs ). This way I don't have to instantiate each migration and call the `check` method. All that can be done from the `MigrationRunner` class.

Reply:
No, the `check` method has nothing to do with dependencies. The migration system uses dates (the timestamp) to avoid having to get into the complexity of migration dependency by ensuring migrations are run in the correct order by datetime. The `check` method simply gives a way for a specific migration to be checked whether or not it thinks it ran correctly. This would be utilized in the `MigrationsRunner` class after each migration has run.

Reply:
OK, I think I didn't explain very well. The dependency array won't replace the timestamp system, it will just help us to check if dependency migrations have run successfully. 
For example, the new Log system has 3 migrations (CreateNewLogTable, MigrateExistingLogs, and DeleteOldLogTables ). If the `CreateNewLogTable` migration fails, the `MigrateExistingLogs` will fail too.  However, the  `DeleteOldLogTables` will run just fine. But then we have a problem because that migration deleted the old log tables. That's why I think the dependency array might be a good solution. Otherwise, I can run the `check` method of the dependency migration inside migration. Is that what you wanted me to do?

Reply:
Given the direction we're going with the migrations be derived from the logs, un-run logs won't show up. Let's say we have Migrations A, B, and C — which need to run in that order. If Migration A passes but migration B fails, then we would have 2 logs:
- Migration A succeeded
- Migration B failed

Migration C will not have run because its timestamp is after B, and Migration B failed. One thing to note here, which may need to be changed, is that migrations are attempted on every admin request. In this case, for example, Migration B will be attempted every time, resulting in a failure. That will fill the log very quickly. The only trick here is if we have it stop upon failure, we need a way of identifying when we're passing again so that Migration C will be attempted.

You and I may need to hop on a call to think through this together. As I'm writing this out I'm realizing there are some tricky nuances to be thought out.

Reply:
Of course, it throws an exception which stops running other migrations. Let's hop on the call today. In the meantime, I'll start working on the interface.

Reply:
Resolving this as we agreed to remove the check method entirely.